classification/keras/code/keras_script.py

```diff
@@ -0,0 +1,482 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Text classification using Keras with Neptune tracking
+# Notebook inspired from https://keras.io/examples/nlp/text_classification_from_scratch/
+
+# ## Setup
+
+# In[41]:
+
+
+import os
+
+import numpy as np
+import tensorflow as tf
+
+# (Neptune) Import Neptune and initialize a project
+
+# In[42]:
+
+
+os.environ["NEPTUNE_PROJECT"] = "showcase/project-text-classification"
+
+
+# In[43]:
+
+
+import neptune.new as neptune
+
+project = neptune.init_project()
+
+
+# ## Data preparation
+# We are using the IMDB sentiment analysis data available at https://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz. For the purposes of this demo, we've uploaded this data to S3 at https://neptune-examples.s3.us-east-2.amazonaws.com/data/text-classification/aclImdb_v1.tar.gz and will be downloading it from there.
+
+# ### (Neptune) Track datasets using Neptune
+# Since this dataset will be used among all the runs in the project, we track it at the project level
+
+# In[4]:
+
+
+project["data/keras/files"].track_files(
+    "s3://neptune-examples/data/text-classification/aclImdb_v1.tar.gz"
+)
+project.sync()
+
+
+# ### (Neptune) Download files from S3 using Neptune
+
+# In[5]:
+
+
+print("Downloading data...")
+project["data/keras/files"].download("..")
+
+
+# ### Extract downloaded files
+
+# In[6]:
+
+
+import tarfile
+
+print("Extracting data...")
+my_tar = tarfile.open("../aclImdb_v1.tar.gz")
+my_tar.extractall("..")
+my_tar.close()
+
+
+# ### Remove `unsup` subfolder and rename `aclImdb` folder
+
+# In[7]:
+
+
+import shutil
+
+shutil.rmtree("../aclImdb/train/unsup")
+os.remove("../aclImdb_v1.tar.gz")
+
+if os.path.exists("../data"):
+    shutil.rmtree("../data")
+os.rename("../aclImdb", "../data")
+
+
+# (Neptune) Upload dataset sample to Neptune project
+
+# In[8]:
+
+
+import random
+
+project["data/keras/sample/train/pos"].upload(
+    f"../data/train/pos/{random.choice(os.listdir('../data/train/pos'))}"
+)
+project["data/keras/sample/train/neg"].upload(
+    f"../data/train/neg/{random.choice(os.listdir('../data/train/neg'))}"
+)
+project["data/keras/sample/test/pos"].upload(
+    f"../data/test/pos/{random.choice(os.listdir('../data/test/pos'))}"
+)
+project["data/keras/sample/test/neg"].upload(
+    f"../data/test/neg/{random.choice(os.listdir('../data/test/neg'))}"
+)
+
+
+# ### Generate training, validation, and test datasets
+
+# In[44]:
+
+
+data_params = {
+    "batch_size": 32,
+    "validation_split": 0.2,
+    "max_features": 20000,
+    "embedding_dim": 128,
+    "sequence_length": 500,
+    "seed": 42,
+}
+
+
+# (Neptune) Log data metadata to Neptune
+
+# In[45]:
+
+
+run = neptune.init_run(name="Keras text classification", tags=["keras"])
+
+
+# In[46]:
+
+
+run["data/params"] = data_params
+
+
+# In[47]:
+
+
+import re
+import string
```

Comment:
All imports to the top would be nice.

Reply:
the script is an export of the notebook.. haven't changed the structure as this format lets you run a script as a notebook in vscode

Reply:
Since we are already adding a NB separately, I think it would be better to showcase the practices using script.